torch/utils/data/datapipes/iter/callable.py

```diff
@@ -68,13 +68,19 @@ def _apply(self, data, nesting_level):
         if nesting_level == 0:
             return self.fn(data, *self.args, **self.kwargs)
         elif nesting_level > 0:
-            if not isinstance(data, list):
+            if isinstance(data, DataChunk):
+                result = data.__class__([self._apply(i, nesting_level - 1) for i in data.raw_iterator()])
```

Comment:
nit: It's better to use `type(data)` than `data.__class__`.
Here is an example: https://stackoverflow.com/a/10633356

Reply:
👍🏻